Fix compatibility with server-side execution extensions (#16764)
jupyterlab/jupyterlab: docs/source/extension/extension_migration.rst

See the `Playwright 1.46.0 release notes <https://github.com/microsoft/playwright/releases/tag/v1.46.0>`_ for more information.

Shared model
~~~~~~~~~~~~

The outputs set on the shared cell model are now expected to be wrapped

> in the pycrdt ``Map`` objects rather than provided as plain dictionaries.

Author: ```suggestion
in the pycrdt ``Map`` objects rather than provided as plain dictionaries.
Further, the `"text"` entry can now be specified as an `Array<string>`
object for outputs of `"stream"` type, allowing for better performance.
```

The use of dictionaries is deprecated and will stop working in a future version.
For reference, see PRs:

- https://github.com/jupyterlab/jupyterlab/pull/16498

> - https://github.com/jupyter-server/jupyter_ydoc/pull/241

Author: @davidbrochart is this correct? Or should we make this an implementation detail and fix this on `jupyter_ydoc` level by modifying the setter of `outputs` to do the wrapping for the API user? It may be hard because I think cells are just `YMap` for now, but I guess for sake of better typing we might want to have a subclass with custom setters - is this possible?

For reference I also opened https://github.com/jupyter-server/jupyter_ydoc/issues/269 to track typing enhancements.

Reviewer: This is correct, but not complete. The reason each output is now a `Y.Map<any>` is that if the output is of type `"stream"`, its `"text"` entry is of type `Y.Array<string>`. This is also why it's difficult to type an output more precisely than `Y.Map<any>`: the type is dynamic, as it depends on the output's type which is only known at run-time. I agree that we could at least type it as a union of two types, one corresponding to a `"stream"` output type and the other to other types of outputs.

See the `Playwright 1.46.0 release notes <https://github.com/microsoft/playwright/releases/tag/v1.46.0>`_ for more information.

Shared model
~~~~~~~~~~~~

The outputs set on the shared cell model are now expected to be wrapped

> in the pycrdt ``Map`` objects rather than provided as plain dictionaries.

Reviewer: Should we mention `pycrdt` here? It seems that we are talking from the frontend point of view, so maybe we should mention `Y.Map`?
```suggestion
in the ``Y.Map`` objects rather than provided as plain objects.
```

Author: Hmm... I think both. But yeas, it makes sense to use `Y.Map` notation as this should be clear that it refers to yjs/pycrdt stack.

Author: ```suggestion
in the ``Y.Map`` objects rather than provided as plain objects
(or pycrdt ``Map`` objects rather than dictionaries when set on the backend).
```

My point is that there is usage of documents similar to server-side execution in the wild.

> Further, the ``"text"`` entry can now be specified as an ``Array<string>``

Reviewer: ```suggestion
Further, the ``"text"`` entry must now be specified as an ``Array<string>``
```

object for outputs of ``"stream"`` type, allowing for better performance.

> The use of dictionaries is deprecated and will stop working in a future version.

Reviewer: ```suggestion
The use of plain objects is deprecated and will stop working in a future version.
```
